assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -106,9 +109,11 @@ class PageEdit extends Component {
 
 			mediaType = media.type;
 		}
+		const fetchedMedia = select( 'core' ).getMedia( media.id );
+		const mediaUrl = has( fetchedMedia, [ 'media_details', 'sizes', MAX_IMAGE_SIZE_SLUG ] ) ? fetchedMedia.media_details.sizes[ MAX_IMAGE_SIZE_SLUG ].source_url : media.url;
```

Comment:
There might be a better way to get the image size than making a new request for images. 

The `media.sizes` array doesn't normally have the `'amp_story_image'` size. Otherwise, this would just get the size from that array.

Reply:
I'll look for a way to have the `'amp_story_image'` size appear in that `sizes` object.

Reply:
It may be just that you have to re-generate image sizes? I found this: https://github.com/WP-API/WP-API/issues/779

Reply:
Nice, thanks for finding that! You'd think that would fix it, but it looks like it's still an issue in my local.

Reply:
Maybe this is not relevant, but at least in the classic Media Library's Ajax actions, there was a `image_size_names_choose` filter for ensuring the additional sizes are included in the response:

https://github.com/WordPress/wordpress-develop/blob/21b3d9031a24ce4821b5ffe613a1926fa08d7380/src/wp-includes/media.php#L3300-L3309

This I don't believe applies to the REST API, however.

Reply:
Yeah, that could have been a really helpful one. It looks like the filter doesn't run in this context, though.

Reply:
Hi @westonruter, it turns out that the `image_size_names_choose` filter that you suggested actually does work. I misspelled the image size slug, so it seemed like it didn't work.

It's used in https://github.com/ampproject/amp-wp/pull/2509/commits/78692f6764900f3a56e3e8d62b4eb97aaa5e7d75, thanks for suggesting it.